For some reason, arm-cx2450x-linux-gnueabi-gcc built with
crosstool-NG 1.3.2 does not like this code and barfs with an
internal compiler error :-(

Index: b/proc/sysinfo.c
===================================================================
--- a/proc/sysinfo.c
+++ b/proc/sysinfo.c
@@ -872,21 +872,21 @@ unsigned get_pid_digits(void){
   char *endp;
   long rc;
   int fd;
   static unsigned ret;
 
-  if(ret) goto out;
+  if(ret) return ret;//goto out;
   ret = 5;
   fd = open("/proc/sys/kernel/pid_max", O_RDONLY);
-  if(fd==-1) goto out;
+  if(fd==-1) return ret;//goto out;
   rc = read(fd, pidbuf, sizeof pidbuf);
   close(fd);
-  if(rc<3) goto out;
+  if(rc<3) return ret;// goto out;
   pidbuf[rc] = '\0';
   rc = strtol(pidbuf,&endp,10);
-  if(rc<42) goto out;
-  if(*endp && *endp!='\n') goto out;
+  if(rc<42) return ret;// goto out;
+  if(*endp && *endp!='\n') return ret; goto out;
   rc--;  // the pid_max value is really the max PID plus 1
   ret = 0;
   while(rc){
     rc /= 10;
     ret++;
